mos_tests/nova/nova_tests.py

```diff
@@ -332,3 +332,62 @@ def test_543356_NovaMassivelySpawnVMsWithBootLocal(self):
             ping = os.system("ping -c 4 -i 8 {}".format(fip_dict[inst_id]))
             msg = "Instance {0} is not reachable".format(inst_id)
             self.assertEqual(ping, 0, msg)
+
+    def test_2238776_NetworkConnectivityToVMDuringLiveMigration(self):
+        """ This test checks
+
+            Steps:
+             1. Create a floating ip
+             2. Create an instance from an image with 'm1.micro' flavor
+             3. Add the floating ip to the instance
+             4. Ping the instance by the floating ip
+             5. Execute live migration
+             6. Check current hypervisor and status of instance
+             7. Check that packets loss was minimal
+
+        """
+        networks = self.neutron.list_networks()['networks']
+        net = [net['id'] for net in networks if not net['router:external']][0]
+        image_id = self.nova.images.list()[0].id
+        flavor = [flavor for flavor in self.nova.flavors.list() if
+                  flavor.name == 'm1.micro'][0]
+        floating_ip = self.nova.floating_ips.create()
+        self.floating_ips.append(floating_ip)
+        self.assertIn(floating_ip.ip, [fip_info.ip for fip_info in
+                                       self.nova.floating_ips.list()])
+        inst = common_functions.create_instance(self.nova, self.instances,
+                                                "inst_2238776_{}"
+                                                .format(flavor.name),
+                                                flavor.id, net,
+                                                self.sec_group.name,
+                                                image_id=image_id)
+        inst.add_floating_ip(floating_ip.ip)
+        ping = os.system("ping -c 4 -i 4 {}".format(floating_ip.ip))
+        self.assertEqual(ping, 0, "Instance is not reachable")
+        hypervisors = {h.hypervisor_hostname: h for h in self.nova.hypervisors.
```

Comment:
hypervisors = {h.hypervisor_hostname: h for h
____________in self.nova.hypervisors.list()}